### PR TITLE
Google benchmark framework and scripts to run benchmark

### PR DESCRIPTION
**Pull request by srsuryadev**
### Summary
In this PR, 
 * Used google's benchmark framework to run the ssb benchmark workload.
 * Also, added scripts to generate data and run the benchmark. 

### Test
Ran the benchmark with google framework using the newly added script,

Screenshot 2020-09-08 at 21 56 59

### CODE REVIEW

**Comment by johnchronis** [comment]
This is a great PR! 

What do you think abou logging the results of each run of the benchmark (maybe for each PR), so we know if performance is improving or not overtime. 

I am not aware of an automatic way to do that, and obviously we would need to keep the hardware we run the benchmark on consistent. I was thinking we could use the hustle server and script that process. Maybe have a plot that tracks progress as well :)

Of course this does not need to happen in this PR, just wanted to discuss!

**Review by johnchronis — APPROVED** [review]

File: scripts/ssb/run_benchmark.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+cur_dir=`dirname $0`
+src_dir=${cur_dir}/../..
+
+# run the benchmark
+cd ${src_dir}/build/src/benchmark/
+./hustle_src_benchmark_main
```
Comment (johnchronis) [inline scripts/ssb/run_benchmark.sh:8]: add newline?

File: src/benchmark/main.cc [hunk 2]
```diff
@@ -107,131 +112,128 @@ void read_from_csv() {
...
+  std::cout << "Started initializing with the required data ..." << std::endl;
+  read_from_csv();
+
+  workload = new SSB();
+
+  ::benchmark::Initialize(&argc, argv);
+
+  std::cout << "Stated running benchmarks ..." << std::endl;
+  ::benchmark::RunSpecifiedBenchmarks();
 }
```
Comment (johnchronis) [inline src/benchmark/main.cc:239]: add newline?

**Review by johnchronis — APPROVED** [review]

**Comment by srsuryadev** [comment]
@johnchronis Yes, I agree with you that it will be greatly helpful. These minor work will become a building block for that. I am also thinking about it.

**Comment by pateljm** [comment]
Nice @srsuryadev!

**Pull request merged**